frontend/packages/shared/src/components/DateTime/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,43 @@
+import { Textfield, TextfieldProps } from '@digdir/design-system-react';
+import React, { ReactNode } from 'react';
+
+export type DatePickerProps = {
+  value: Date;
+  onChange: (date: Date) => void;
+} & Omit<TextfieldProps, 'onChange' | 'value'>;
+
+/**
+ * @component
+ *    Displays a Date picker where the user can select a new date.
+ *
+ * @example
+ *    const [date, setDate] = useState(new Date('2023-10-12'));
+ *
+ *    <DatePicker
+ *      value={date}
+ *      onChange={(newDate: Date) => setDate(newDate)}
+ *      label='Select date'
+ *      size='small'
+ *    />
+ *
+ * @property {Date}[value] - The date to display
+ * @property {function}[onChange] - Function that updates the date value
+ * @property {string}[label] - The label of the component
+ * @property {'xsmall' | 'small' | 'medium' | 'large'}[value] - the size of the component
+ *
+ * @returns {ReactNode} - The rendered component
+ */
+export const DatePicker = ({ value, onChange, ...rest }: DatePickerProps): ReactNode => {
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    onChange(new Date(e.target.value));
```

Comment:
Egentlig er jeg usikker på om vi burde bruke `Date` her. `Date` vil alltid referere til kl. 00:00 UTC når man ikke har spesifisert klokkeslett, så brukeren kan risikere å få feil dato pga. tidssoneforskyvninger, i hvert fall hvis vedkommende befinner seg på vestlig side av Greenwich. Det eneste vi er interessert i her er år, måned og dag, så jeg foreslår at vi returnerer strengen direkte på samme måte som i `TimePicker`.

Reply:
Det er et veldig vesentlig poeng du kommer med her @TomasEng. 👍  Personlig hadde jeg forventet å få tilbake en Date her og at jeg selv må passe på å bruke `Intl.DateFormat` eller `Date().toLocaleString({ timezone-options })` for å formatere den riktig. Det gir også meg som utvikler større frihet til hva jeg ønsker å gjøre med resultatet fra DatePicker. Hva tenker du om det? 😊

Reply:
Ja, sånn umiddelbart ville jeg også tenkt det samme. Men det er ikke formatet jeg synes er problematisk. Problemet er at `Date` automatisk legger til mer informasjon enn det brukeren har fylt ut, og det kan føre til konverteringsproblemer.

Eksempel:
1. En bruker som befinner seg i New York fyller ut dagens dato, 13.10.2023.
2. Datoen blir konvertert til Date. Da er det ikke lengre bare en dato, men et nøyaktig tidspunkt, nemlig 13.10.2023 kl. 00:00:00 UTC (som er det samme som 12.10.2023 kl. 20:00 i New York).
3. Appen behandler verdien uten å ta hensyn til informasjonen om tidssone (naturlig nok siden det ikke er noe den har bedt om) og ender opp med å lagre datoen 12.10.2023, og/eller oppdatere inputfeltet med denne.

Dette er en type feil som det er vanskelig å oppdage siden den ikke vil oppstå før noen tar i bruk appen i en annen verdensdel. Det er heller ikke intuitivt at man må ta hensyn til klokkeslett.
Å returnere strengen direkte gjør ikke friheten noe mindre - tvert imot gjør det at utvikleren selv kan vurdere hvorvidt det er nødvendig å konvertere datoen til `Date`, og deretter ta hensyn til forholdsreglene som kreves.

Reply:
For å ta en annen demo av det samme problemet, se hva man får når man kjører denne:
```
new Date('2023-10-13').toLocaleDateString(undefined, {timeZone: 'EST'})
```